src/js/plugins/plugin.search.js

```diff
@@ -0,0 +1,599 @@
+ */
+
+/**
+ * Submits search request
+ *
+ * @param {string} term
+ * @param {searchOptions} options
+ */
+BookReader.prototype.search = function(term, options) {
+  options = options !== undefined ? options : {};
+  /** @type {searchOptions} */
+  const defaultOptions = {
+    goToFirstResult: false, /* jump to the first result (default=false) */
+    disablePopup: false,    /* don't show the modal progress (default=false) */
+    error: null,            /* optional error handler (default=null) */
+    success: null,          /* optional success handler (default=null) */
+  };
+  options = jQuery.extend({}, defaultOptions, options);
+
+  this.$('.BRsearchInput').blur(); //cause mobile safari to hide the keyboard
+
+  this.removeSearchResults();
+
+  this.searchTerm = term;
+  // strip slashes, since this goes in the url
+  this.searchTerm = this.searchTerm.replace(/\//g, ' ');
+
+  this.trigger(BookReader.eventNames.fragmentChange);
+
+  // Add quotes to the term. This is to compenstate for the backends default OR query
+  // term = term.replace(/['"]+/g, '');
+  // term = '"' + term + '"';
+
+  // Remove the port and userdir
+  const serverPath = this.server.replace(/:.+/, '');
+  const baseUrl = `https://${serverPath}${this.searchInsideUrl}?`;
+
+  // Remove subPrefix from end of path
+  let path = this.bookPath;
+  const subPrefixWithSlash = `/${this.subPrefix}`;
+  if (this.bookPath.length - this.bookPath.lastIndexOf(subPrefixWithSlash) == subPrefixWithSlash.length) {
+    path = this.bookPath.substr(0, this.bookPath.length - subPrefixWithSlash.length);
+  }
+
+  const urlParams = {
+    item_id: this.bookId,
+    doc: this.subPrefix,
+    path,
+    q: term,
+  };
+
+  // NOTE that the API does not expect / (slashes) to be encoded. (%2F) won't work
+  const paramStr = $.param(urlParams).replace(/%2F/g, '/');
+
+  const url = `${baseUrl}${paramStr}`;
+
+  if (!options.disablePopup) {
+    this.showProgressPopup(
+      `<img class="searchmarker" src="${this.imagesBaseURL}marker_srch-on.svg" />
+      Search results will appear below...`);
+  }
+
+  const processSearchResults = (searchInsideResults) => {
+    const responseHasError = searchInsideResults.error || !searchInsideResults.matches.length;
+    const hasCustomError = typeof options.error === 'function';
+    const hasCustomSuccess = typeof options.success === 'function';
+
+    if (responseHasError) {
+      hasCustomError
+        ? options.error(searchInsideResults, options)
+        : this.BRSearchCallbackError(searchInsideResults, options);
+    } else {
+      hasCustomSuccess
+        ? options.success(searchInsideResults, options)
+        : this.BRSearchCallback(searchInsideResults, options);
+    }
+  };
+
+  $.ajax({
+    url: url,
+    dataType: 'jsonp'
+  }).then(processSearchResults);
+};
+
+/**
+ * Search Inside results shape
+ * @typedef {object} searchInsideResults
+ * @property {string} error
+ * @property {array} matches
+ * @property {boolean} indexed
+ */
```

Comment:
This should get all of them :+1:

```suggestion
/**
  * @typedef {object} SearchInsideMatchBox
  * @property {number} page
  * @property {number} r
  * @property {number} l
  * @property {number} b
  * @property {number} t
  * @property {HTMLDivElement} [div]
  */

/**
 * @typedef {object} SearchInsideMatch
 * @property {string} text
 * @property {Array<{ page: number, boxes: SearchInsideMatchBox[] }>} par
 */

/**
 * @typedef {object} SearchInsideResults
 * @property {string} error
 * @property {SearchInsideMatch[]} matches
 * @property {boolean} indexed
 */
```